hazelcast/src/main/java/com/hazelcast/datalink/DataLink.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datalink;
+
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.spi.annotation.Beta;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
```

Comment:
Would the name 'Connector' not make more sense? That is the name used in e.g. Kafka.

https://kafka.apache.org/documentation/#connectconfigs

Reply:
Re: naming, the terminology list is described in TDD : https://github.com/hazelcast/hazelcast/blob/master/docs/design/sql/18-data-links.md#terminology

Reply:
From the documentation it isn't clear what the relation is between a connector and a Datalink.

Reply:
I have updated the javadoc for both `DataLink` and `DataLinkService`. See if it is clearer now.

Reply:
It still isn't clear to me. If you can give a concrete example, then hopefully the seperation will make more sense.

Reply:
Ok. It is a bit cleared to me now. In terms ofJDBC a datalink could either be a shared connection or even a connection pool.